pkg/nodebootstrap/reserved.go

```diff
@@ -93,17 +98,17 @@ func (i InstanceTypeInfo) DefaultStorageToReserve() string {
 
 // DefaultMemoryToReserve returns how much memory to reserve.
 //
-// See https://github.com/awslabs/amazon-eks-ami/blob/ff690788dfaf399e6919eebb59371ee923617df4/files/bootstrap.sh#L150-L181
-// which takes it form https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-architecture#node_allocatable
-//
-// 255 Mi of memory for machines with less than 1024Mi of memory
-// 25% of the first 4096Mi of memory
-// 20% of the next 4096Mi of memory (up to 8192Mi)
-// 10% of the next 8192Mi of memory (up to 16384Mi)
-// 6% of the next 114688Mi of memory (up to 131072Mi)
-// 2% of any memory above 131072Mi
+// See https://github.com/awslabs/amazon-eks-ami/blob/21426e27e3845319dbca92e7df32e5c4b984a1d1/files/bootstrap.sh#L154-L165
 func (i InstanceTypeInfo) DefaultMemoryToReserve() string {
-	mib := memProgression.calculate(i.Memory, minimumMemoryToReserve)
+	var mib int64
+	if i.MaxPodsPerNode > 0 {
+		mib = 11*i.MaxPodsPerNode + minimumMemoryToReserve
+	} else {
+		// As a fallback, if MaxPodsPerNode is 0, use the original calculation, which is a progressive calculation based on total memory.
```

Comment:
Do we need the fallback? I don't see any types with `MaxPodsPerNode: 0`

Reply:
You probably saw this, but in the diff, it is collapsed. Some examples of instance types with no max pods are:
- m5ad.8xlarge
- m5ad.16xlarge

I wasn't really quite sure what to make of this, as these seem to not be available when I created an EC2 instance via the console. They don't exist in the list.

I was hesitant to completely remove the old calc in case those instance types need to be supported.

As a test, I just tried to create a cluster using m5ad.8xlarge, and it failed, so I guess those instance types won't work anyway, so I think you're probably right, the fallback is unnecessary and can be removed.

I'll update the PR to remove it.

Reply:
Ah yeah true, I was checking for a missing line `MaxPodsPerNode:`.

Reply:
Ok, those instances don't exist apparently (anymore?). Now I'm wondering why they show up at all.

Reply:
I pushed a new update that removes the fallback and related field which is no longer used.

I also discovered a problem (with this new PR) where there are mixed instance types and it takes the minimum across all types.  I had not updated that branch of the code, so it was using the fallback, and the unit test still passed because it was using the fallback calc.

ptal and see what you think now.